File: generate-icons.js
#!/usr/bin/env node
/**
 * Icon generator using Sharp
 * Converts SVG icon to multiple PNG sizes for favicon and app icons
 */

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const SIZES = [
  { size: 512, name: 'icon-512.png', desc: 'App icon' },
  { size: 192, name: 'icon-192.png', desc: 'Android icon' },
  { size: 96, name: 'icon-96.png', desc: 'Icon' },
  { size: 32, name: 'favicon.png', desc: 'Favicon' },
  { size: 16, name: 'favicon-16.png', desc: 'Favicon 16x16' },
];

const SVG_PATH = path.join(__dirname, 'public', 'icon.svg');
const OUTPUT_DIR = path.join(__dirname, 'public');

async function generateIcons() {
  try {
    console.log('🎨 Starting icon generation from SVG...\n');

    if (!fs.existsSync(SVG_PATH)) {
      throw new Error(`SVG file not found: ${SVG_PATH}`);
    }

    // Ensure output directory exists
    if (!fs.existsSync(OUTPUT_DIR)) {
      fs.mkdirSync(OUTPUT_DIR, { recursive: true });
    }

    // Read SVG
    const svgBuffer = fs.readFileSync(SVG_PATH);

    // Generate each size
    for (const { size, name, desc } of SIZES) {
      const outputPath = path.join(OUTPUT_DIR, name);
      process.stdout.write(`  📐 Generating ${size}x${size} (${desc})... `);

      await sharp(svgBuffer)
        .resize(size, size, {
          fit: 'contain',
          background: { r: 0, g: 0, b: 0, alpha: 0 },
        })
        .png({ quality: 90 })
        .toFile(outputPath);

      const stats = fs.statSync(outputPath);
      console.log(`✅ ${name} (${(stats.size / 1024).toFixed(2)} KB)`);
    }

    console.log('\n✨ Icon generation complete!');
    console.log(`\n📦 Generated files in ${OUTPUT_DIR}:`);
    for (const { name } of SIZES) {
      const fullPath = path.join(OUTPUT_DIR, name);
      if (fs.existsSync(fullPath)) {
        const stats = fs.statSync(fullPath);
        console.log(`   ✓ ${name} (${(stats.size / 1024).toFixed(2)} KB)`);
      }
    }

  } catch (error) {
    console.error('❌ Error generating icons:', error.message);
    process.exit(1);
  }
}

generateIcons();

File: generate_icons.py
#!/usr/bin/env python3
"""
Icon generator - creates PNG and favicon files from SVG
"""

import os
import subprocess
import sys

# Check if cairosvg is installed, if not try to install it
try:
    import cairosvg
except ImportError:
    print("Installing cairosvg for SVG conversion...")
    subprocess.check_call([sys.executable, "-m", "pip", "install", "cairosvg", "pillow"])
    import cairosvg

from PIL import Image

def generate_icons():
    """Generate all icon sizes from SVG"""
    svg_path = "public/icon.svg"
    output_dir = "public"
    
    if not os.path.exists(svg_path):
        print(f"Error: {svg_path} not found!")
        return False
    
    print(f"📐 Reading SVG from {svg_path}")
    
    # Read SVG content
    with open(svg_path, 'r') as f:
        svg_content = f.read()
    
    # Define icon sizes to generate
    sizes = [
        (512, "icon-512.png"),
        (192, "icon-192.png"),
        (96, "icon-96.png"),
        (32, "favicon.png"),
        (16, "favicon-16.png"),
    ]
    
    print(f"🎨 Generating icon sizes: {', '.join([f'{size[0]}x{size[0]}' for size in sizes])}")
    
    # Generate PNG files
    for size, filename in sizes:
        output_path = os.path.join(output_dir, filename)
        print(f"  Converting to {size}x{size}... {filename}")
        
        try:
            # Convert SVG to PNG using cairosvg
            cairosvg.svg2png(bytestring=svg_content.encode('utf-8'), 
                           write_to=output_path,
                           output_width=size,
                           output_height=size)
            print(f"  ✅ Created {filename}")
        except Exception as e:
            print(f"  ❌ Failed to create {filename}: {e}")
            return False
    
    print(f"\n✨ Icon generation complete!")
    print(f"Files created in {output_dir}:")
    for _, filename in sizes:
        path = os.path.join(output_dir, filename)
        if os.path.exists(path):
            size = os.path.getsize(path)
            print(f"  ✓ {filename} ({size} bytes)")
    
    return True

if __name__ == "__main__":
    success = generate_icons()
    sys.exit(0 if success else 1)
